### PR TITLE
Feat/signers read stackerdb

### PR DESCRIPTION
**Pull request by jferrant**
Closes https://github.com/stacks-network/stacks-core/issues/4636

Attempted to do for sign as well, but this would require substantial changes to the way a miner stores its messages as we require every single sign message to replicate state and miners overwrite their own messages as they use the same slot for everything. Sign rounds are also quite quick and a miner already has timeout logic in place for triggering a new sign request, so this is less of an issue anyway.

### CODE REVIEW

File: .github/workflows/bitcoin-tests.yml
Comment (jferrant) [inline .github/workflows/bitcoin-tests.yml:85]: Note: this is just a rename as the test does not actually perform sign anymore (changed in a different PR)

**Review by kantai — APPROVED** [review]
This code LGTM! I'm curious about the slow test, though. It doesn't seem like restarting the signer should require re-syncing the stacks-node.

**Comment by jferrant** [comment]
> This code LGTM! I'm curious about the slow test, though. It doesn't seem like restarting the signer should require re-syncing the stacks-node.

Yeah, I am not 100% sure why this is the case. I actually can't merge this until https://github.com/stacks-network/stacks-core/issues/4654 is done. So will keep this as is for now and will look into it further when that PR is completed.

File: stacks-signer/src/signer.rs
Comment (hstove) [inline stacks-signer/src/signer.rs:1465]: Can you explain this code a little bit? I have no reason to doubt it, I'm just not totally following this code path

File: testnet/stacks-node/src/tests/signer.rs
Comment (hstove) [inline testnet/stacks-node/src/tests/signer.rs:1184]: Won't this never happen, even if you didn't stop a signer, because you haven't advanced the Stacks tip? So the `vote-for-aggregate-key` votes wouldn't be confirmed here?

Also because, if I'm reading this right, we're still in 2.5, so there wouldn't be Nakamoto blocks.

File: testnet/stacks-node/src/tests/signer.rs
Comment (hstove) [inline testnet/stacks-node/src/tests/signer.rs:1117]: This is a really good test!

**Review by hstove — APPROVED** [review]
I know it's in draft, but I had already started a review, so I figured I'd just finish it.

File: stacks-signer/src/signer.rs
Comment (jferrant) [reply stacks-signer/src/signer.rs:1465]: Yeah this is because I was working on a different PR simultaneously (see https://github.com/stacks-network/stacks-core/pull/4660) It is one of the reasons I converted it to a draft so I can rebase against it once merged :) 

Basically this is to handle the fact that we could trigger a DKG round due to timing issues where we might have an outdated view (for example, exceeded the threshold check but still not gotten the approved key though the other PR will mitigate this). If this happens and then we later discover that DKG was triggered and it shouldn't have been, we should abort our attempt. This is incomplete however until we can then load the appropriate DKG shares from storage for the winning round. Hence the link to Marten's issue. So I have put this ticket a bit on pause until I can build off these two other PRs.

File: testnet/stacks-node/src/tests/signer.rs
Comment (jferrant) [reply testnet/stacks-node/src/tests/signer.rs:1184]: Oh good point. I should mine one block before to prove that DKG did not proceed. Will add. And yes this is 2.5. I don't make any references to Nakamoto blocks in this test for that reason.

**Review by hstove — DISMISSED** [review]

File: testnet/stacks-node/src/tests/signer.rs
Comment (xoloki) [inline testnet/stacks-node/src/tests/signer.rs:1126]: It's so exciting that this works now!!!

**Review by xoloki — APPROVED** [review]
Looks great! So exciting to get this in, we've talked about it forever.

File: stacks-signer/src/signer.rs
Comment (hstove) [inline stacks-signer/src/signer.rs:1412]: Is this TODO still relevant? From my read it seems like it is?

In that case, does the later `finish_operation` command essentially prevent the signer from voting on that key?

Comment (hstove) [reply stacks-signer/src/signer.rs:1412]: Also, if my read is right, I think the `warn` when updating DKG should be an `error` with something more explicit (like "didn't participate in DKG, unable to sign during this reward cycle")

**Review by hstove — APPROVED** [review]
Nice job. A little nit regarding comments / logs, and that may be to me misunderstanding things. It shouldn't be blocking a merge, but a few more comments around why this state happened in refresh/update DKG would be help.

Also, I was going to comment "nice job on that test" but I already did 😂

File: stacks-signer/src/signer.rs
Comment (jferrant) [reply stacks-signer/src/signer.rs:260]: It shouldn't unless we have a malicious or buggy signer.

File: stacks-signer/src/signer.rs
Comment (jferrant) [reply stacks-signer/src/signer.rs:1424]: No this message is not stale. You can set the DKG key from the contract while in the middle of a round and in which case you should abort that round. (this happens if a coordinator thinks DKG round has failed and a new one gets triggered)

**Review by jcnelson — APPROVED** [review]

**Pull request merged**
